Pay/PaySession.swift

```diff
@@ -27,33 +27,36 @@
 import Foundation
 import PassKit
 
-public protocol PayControllerDelegate: class {
-    func payController(_ payController: PayController, didRequestShippingRatesFor address: PayPostalAddress, provide: @escaping (PayCheckout, [PayShippingRate]) -> Void)
-    func payController(_ payController: PayController, didSelectShippingRate shippingRate: PayShippingRate, provide: @escaping  (PayCheckout) -> Void)
-    func payController(_ payController: PayController, didCompletePayment token: Data, checkout: PayCheckout, completeTransaction: @escaping (PayController.TransactionStatus) -> Void)
-    func payControllerDidFinish(_ payController: PayController)
+public protocol PaySessionDelegate: class {
+    func paySession(_ paySession: PaySession, didRequestShippingRatesFor address: PayPostalAddress, checkout: PayCheckout, provide: @escaping (PayCheckout?, [PayShippingRate]) -> Void)
+    func paySession(_ paySession: PaySession, didSelectShippingRate shippingRate: PayShippingRate, checkout: PayCheckout, provide: @escaping (PayCheckout?) -> Void)
+    func paySession(_ paySession: PaySession, didAuthorizePayment authorization: PayAuthorization, checkout: PayCheckout, completeTransaction: @escaping (PaySession.TransactionStatus) -> Void)
+    
+    func paySessionDidFinish(_ paySession: PaySession)
 }
 
-public class PayController: NSObject {
+public class PaySession: NSObject {
     
     public enum TransactionStatus {
         case success
         case failure
     }
     
-    public weak var delegate: PayControllerDelegate?
+    public weak var delegate: PaySessionDelegate?
     
     public let merchantID: String
+    public let identifier: String
     
-    public fileprivate(set) var currency:      PayCurrency?
-    public fileprivate(set) var checkout:      PayCheckout?
-    public fileprivate(set) var shippingRates: [PayShippingRate]?
+    internal fileprivate(set) var currency:      PayCurrency?
+    internal fileprivate(set) var checkout:      PayCheckout?
```

Comment:
So in this case you're totally right, if it's `nil` then the sky is falling and we should probably crash.

My (more general) concern is whoever will next touch this code (no idea who that might be 😉) may not check the variable declaration, assume that this implicitly-unwrapped variable is safe to use anywhere, try to access it before it's initialized, and crash. If `!`s are always avoided, the compiler will never let that happen.

That all having been said, I definitely don't feel super strongly about this. 👍